TombEngine/Objects/TR3/Entity/tr3_civvy.cpp

```diff
@@ -2,73 +2,113 @@
 #include "Objects/TR3/Entity/tr3_civvy.h"
 
 #include "Game/control/box.h"
+#include "Game/control/lot.h"
 #include "Game/effects/effects.h"
-#include "Game/Lara/lara.h"
-#include "Game/itemdata/creature_info.h"
 #include "Game/items.h"
+#include "Game/itemdata/creature_info.h"
+#include "Game/Lara/lara.h"
 #include "Game/misc.h"
 #include "Game/people.h"
+#include "Math/Random.h"
 #include "Sound/sound.h"
 #include "Math/Math.h"
+
 #include "Specific/level.h"
 #include "Specific/setup.h"
 
 using namespace TEN::Math;
+using std::vector;
 
 namespace TEN::Entities::Creatures::TR3
 {
-	constexpr auto CIVVY_ATTACK_DAMAGE = 40;
-	constexpr auto CIVVY_SWIPE_DAMAGE  = 50;
+	constexpr auto CIVVY_ATTACK_DAMAGE  = 50;
 
-	constexpr auto CIVVY_ATTACK0_RANGE = SQUARE(BLOCK(3));
-	constexpr auto CIVVY_ATTACK1_RANGE = SQUARE(BLOCK(0.67f));
-	constexpr auto CIVVY_ATTACK2_RANGE = SQUARE(BLOCK(1));
-	constexpr auto CIVVY_WALK_RANGE	   = SQUARE(BLOCK(1));
-	constexpr auto CIVVY_ESCAPE_RANGE  = SQUARE(BLOCK(3));
-	constexpr auto CIVVY_AWARE_RANGE   = SQUARE(BLOCK(1));
+	constexpr auto CIVVY_ATTACK_WALKING_PUNCH_RANGE = SQUARE(BLOCK(0.75f));
+	constexpr auto CIVVY_ATTACK_CLOSE_PUNCH_RANGE = SQUARE(BLOCK(0.45f));
+	constexpr auto CIVVY_ATTACK_FAR_PUNCH_RANGE = SQUARE(BLOCK(0.70f));
+	
+	constexpr auto CIVVY_WALK_RANGE	   = SQUARE(BLOCK(2));
+	constexpr auto CIVVY_ESCAPE_RANGE  = SQUARE(BLOCK(5));
+	constexpr auto CIVVY_AWARE_RANGE   = SQUARE(BLOCK(15));
 
 	constexpr auto CIVVY_WAIT_CHANCE	   = 1.0f / 128;
-	constexpr auto CIVVY_STATE_WALK_CHANCE = 1.0f / 128; // Unused.
 
 	constexpr auto CIVVY_VAULT_SHIFT = 260;
 
-	constexpr auto CIVVY_WALK_TURN_RATE_MAX = ANGLE(5.0f);
-	constexpr auto CIVVY_RUN_TURN_RATE_MAX	= ANGLE(6.0f);
+	#define CIVVY_WALK_TURN_RATE_MAX ANGLE(5.0f)
```

Comment:
turn this to constexpr auto and move it before CIVVY_ATTACK_WALKING_PUNCH_RANGE.
same for the two below.

Reply:
Done